### DISCUSSION TITLE
Query with a dynamic table

### DISCUSSION THREAD

**Discussion by ploef** [Budibase/budibase#7259] [Help]
Hi,

Would it be possible to create a query with a dynamic table name?

select * from {{ tablename }}

Where table name would be the selected value from a picker

**Comment by R2bEEaton** [comment]
@ploef Hi! You should first ask yourself, "Why am I trying to do this?" Dynamic table names are not a standard operation and this can leave you open to all sorts of problems like code injection, etc. If it's for an internal app with access control, that could be fine, but if there's any way to get around doing this, I'd say try that first.

If you really want to do it, here is a way adapted from [this post](https://stackoverflow.com/questions/2838490/a-table-name-as-a-variable) on Stack Overflow. In Budibase, use the following in your query:

```
EXEC('SELECT * FROM ' + {{ tablename }});
```

where `{{ tablename }}` is a query binding which you will set to a valid table name in your database.

Hope that helps!

**Reply by ploef** [reply]
Hi @R2bEEaton, thanks for the answer! I understand your warning. I have a database with many tables and I'm looking for the best way to let a user select a table from a dropdown and fill a table with the records of the selected table. At the moment I'm filling the dropdown with a SQl query getting all the tablesnames.

**Reply by R2bEEaton** [reply]
That seems like it'd be the way to go. I think what you've got already is a great start and might not need to be changed. With your existing query, the user can pick the table name from the dropdown which could then be sent to this new query and used as a data provider with a default table inside it. The table should then be populated with all of the rows and columns.
